process/transaction/metaProcess.go

```diff
@@ -135,7 +135,13 @@ func (txProc *metaTxProcessor) ProcessTransaction(tx *transaction.Transaction) (
 		return 0, err
 	}
 
+	txCopy := *tx
 	txType, _ := txProc.txTypeHandler.ComputeTransactionType(tx)
+	if txType == process.RelayedTxV3 {
+		// extract the inner transaction in order to get the proper user tx type
+		txCopy.RelayerSignature = nil
```

Comment:
this is an ugly hack. As we have inShard txType, dest txType already returned by compute Transaction type - you could have a 3rd return, or a txType structure return.

Reply:
refactored with extra return, simplifies a lot